internal/sys/fs.go

```diff
@@ -255,6 +255,11 @@ type cachedStat struct {
 	Ino uint64
 }
 
+// IsAppend returns true if the file is open with syscall.O_APPEND.
+func (f *FileEntry) IsAppend() bool {
+	return f.openFlag&syscall.O_APPEND != 0
```

Comment:
this is temporary until the file type is sorted out. Ideally we don't need much of a FileEntry thing later